src/com/tictactoe/tictactoegame/TicTacToeGame.java

```diff
@@ -0,0 +1,212 @@
+            }
+        }
+        return true;
+    }
+
+    /**
+     * places the right symbol in the chosen cell
+     * @param symbol : current player's cell
+     */
+    public void cellPlacement(char symbol){
+        gameMap[line][col]=symbol;
+    }
+
+    /**
+     * checks if game is over by calling two methods victoryChecking() and exAequo();
+     * @param player : Object with attributes symbol and name
+     * @return true if game is over
+     */
+    public boolean endOfGame(Player player ) {
+        if (victoryChecking(player.symbol)){
+            System.out.println("Nous avons un winner! : "+ player.name+ " a gagné!");
+        }
+        return (victoryChecking(player.symbol)) || exAequo();
+    }
+
+    /**
+     * Checks all the possible alignments
+     * @param symbol : current player's symbol
+     * @return true if there is at least one alignment OK (stops at the first found alignment)
+     */
+    public boolean victoryChecking(char symbol){
+        int countSymbol;
+        //horizontal
+        countSymbol=0;
+        for (char charac : gameMap[line]){
+            if (charac==symbol){
+                countSymbol++;
+            }
+        }
+        if (countSymbol==3){
+            return true;
+        }
+        //vertical
+        countSymbol=0;
+        for (char [] line : gameMap){
+            if (line[col]==symbol){
+                countSymbol++;
+            }
+        }
+        if (countSymbol==3){
+            return true;
+        }
+        //first diagonal
+        if (countDiagonal("first",symbol)==3){
+            return true;
+        }
+
+        //second diagonal
+        return(countDiagonal("second",symbol)==3);
+    }
+
+    /**
+     * checks if it remains some empty cells
+     * @return false as soon as it finds an empty cell, either true
+     */
+    public boolean exAequo(){
+        for (char[] line : gameMap){
+            for (char charac : line){
+                if ((charac!='X')&&(charac!='O')){
+                    return false;
+                }
+            }
+        }
+        System.out.println ("Unis dans la défaite! Personne n'a gagné, mais personne n'a perdu non plus ;)");
+        return true;
+    }
+
+   public int lineCalcFunc(int coor){
+        return (int) Math.floor((double) (coor-1)/3);
+   }
+
+    public int colCalcFunc(int coor){
+        return (coor-1)%3;
+    }
+
+    /**
+     * checks diagonal alignments
+     * @param whichDiagonal : specifies what diagonal is about
+     * @param symbol : current player's symbol
+     * @return total symbol number on the diagonal
+     */
+    public int countDiagonal(String whichDiagonal, char symbol){
+        List<Integer> diagonalList;
+        int countSymbol=0;
+
+        if (whichDiagonal.equals("first")){
+            diagonalList=coorFirstDiagPossible;
+        }else {
+            diagonalList=coorSecondDiagPossible;
+        }
```

Comment:
I think we could take the `diagonalList` as parameter instead of this string.

```suggestion
    public int countDiagonal(List<Integer> diagonalList, char symbol){
        List<Integer> diagonalList;
        int countSymbol=0;
```

(and of course change every calls of this method)